cmd/frontend/graphqlbackend/repository.go

```diff
@@ -344,10 +345,38 @@ func (r *RepositoryResolver) Changelist(ctx context.Context, args *RepositoryCha
 		return nil, err
 	}
 
+	// Changelists only exist for perforce repos.
+	if repo.ExternalRepo.ServiceType != extsvc.TypePerforce {
+		return nil, nil
+	}
+
 	rc, err := r.db.RepoCommitsChangelists().GetRepoCommitChangelist(ctx, repo.ID, cid)
 	if err != nil {
+		// Not found could mean either the mapper has some bug, is currently
+		// broken, or hasn't run yet.
+		// We take a slow path here to check if we can find the commit in the repo.
 		if errcode.IsNotFound(err) {
-			return nil, nil
+			r.logger.Debug("failed to find changelist ID - trying to find it via log", log.String("args.CID", args.CID))
+			cs, err := r.gitserverClient.Commits(ctx, r.name, gitserver.CommitsOptions{
+				AllRefs:      true,
+				MessageQuery: fmt.Sprintf("change = %d]", cid),
```

Comment:
small preference: would prefer having dedicate variable name for this (would help with setting breakpoints if I want to)

```suggestion
messageQuery := fmt.Sprintf("change = %d]", cid)
MessageQuery: messageQuery,
```